internal/consumer/service.go

```diff
@@ -0,0 +1,82 @@
+package consumer
+
+import (
+	"log"
+	"runtime"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type Item struct {
+	By          string `bson:"by"`
+	Title       string `bson:"title"`
+	Url         string `bson:"url"`
+	Text        string `bson:"text"`
+	ItemType    string `bson:"itemType" json:"type"`
+	Descendants int    `bson:"descendants"`
+	Id          int    `bson:"id"`
+	Score       int    `bson:"score"`
+	Time        int    `bson:"time"`
+	Parent      int    `bson:"parent"`
+	Poll        int    `bson:"poll"`
+	Kids        []int  `bson:"kids"`
+	Parts       []int  `bson:"parts"`
+	Deleted     bool   `bson:"deleted"`
+	Dead        bool   `bson:"dead"`
+}
+
+func Execute(dbRepo DbRepository, dataService DataService) {
+	ids, err := dataService.getTopStories()
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "Consume: "))
+	}
+
+	idChan := make(chan int)
+	itemChan := make(chan Item)
+
+	go populateIdChan(idChan, ids)
+
+	go fanOutIds(idChan, itemChan, dataService)
+
+	for i := range itemChan {
+		err := dbRepo.SaveItem(i)
+		if err != nil {
+			log.Print(errors.Wrap(err, "Consume: "))
```

Comment:
I'd return the error here and deal with it further up.

Reply:
whats the benefit of doing this? 

Given this is a for loop we could have multiple errors to log (but we don't want to stop the execution), not sure how it could work with bubbling up all the potential errors.